packages/lesswrong/server/repos/VotesRepo.ts

```diff
@@ -109,4 +116,48 @@ export default class VotesRepo extends AbstractRepo<DbVote> {
       WHERE ARRAY_POSITION("authorIds", $1) IS NOT NULL
     `, [oldUserId, newUserId]);
   }
+
+  // Get votes from recent content by a user,
+  // to use to decide on their rate limit
+  // (note: needs to get the user's own self-upvotes so that
+  // it doesn't skip posts with no other votes)
+  async getVotesOnRecentContent(userId: string): Promise<RecentVoteInfo[]> {
+    const voteFields = `"Votes"._id, "Votes"."userId", "Votes"."power", "Votes"."documentId", "Votes"."collectionName", `
+    const votes = await this.any(`
+      (
+        SELECT ${voteFields} "Posts"."postedAt"
+        FROM "Votes"
+        JOIN "Posts" on "Posts"._id = "Votes"."documentId"
+        WHERE
+          "Votes"."documentId" in (
+            SELECT _id FROM "Posts" 
+            WHERE
+              "Posts"."userId" = '${userId}'
+            ORDER BY "Posts"."postedAt" DESC
+            LIMIT ${RECENT_CONTENT_COUNT}
+          )
+          AND 
+          "cancelled" IS NOT true
+        ORDER BY "Posts"."postedAt" DESC
+      )
+      UNION
+      (
+        SELECT ${voteFields} "Comments"."postedAt"
+        FROM "Votes"
+        JOIN "Comments" on "Comments"._id = "Votes"."documentId"
+        WHERE
+          "Votes"."documentId" in (
+            SELECT _id FROM "Comments" 
+            WHERE
+              "Comments"."userId" = '${userId}' 
+            ORDER by "Comments"."postedAt" DESC
+            LIMIT ${RECENT_CONTENT_COUNT}
+          )
+          AND
+          "cancelled" IS NOT true
+        ORDER BY "Comments"."postedAt" DESC
+      )
+    `)
+    return votes as RecentVoteInfo[]
```

Comment:
This isn't necessary, you should just be able to `return this.any` and the return type annotation on `getVotesOnRecentContent` will take care of it (since `this.any` already returns a `Promise<any[]>`)

Reply:
It yells at me if I don't include the `as RecentVoteInfo[]`. 

As noted elsewhere, I could use help figuring out exactly how the typing is supposed to work here.